### PR TITLE
Add Sound page to the Programming Guide

### LINKED ISSUE #1274: Documentation: Sound compatibility recommendations
### What documentation needs to  be added? 
Information on maximizing cross-platform compatibility for sound assets.

### Where should it be added?
Probably the programming guide.

### Details
Switching from soloud to pyglet for sound means that we inherit pyglet's issues. On OpenAL backends, only 16 bit audio is supported. Linux users can switch to a pulse backend, but macs reportedly only support 16 bit:
> Your audio files are 24bit, but unfortunately the OpenAL backend does not support that. 
> Linux also has a PulseAudio backend available, but for OSX it only has OpenAL.
> Essentially you need to use 16bit audio files for cross platform support.

A user ran into this on the pyglet discord server, and our users may be more confused by this as beginners. An easy recommendation would be to import assets into Audacity and export as 16 bit PCM wav, which is the default setting for wav export.

### LINKED ISSUE #1900: Document PulseAudio crash + workarounds
## Documentation request:

tl;dr pyglet's PulseAudio backend can crash but using `libopenal` fixes it

### Where should it be located?

The Linux install instructions

### What needs to be added

1. A brief summary of https://github.com/pyglet/pyglet/issues/952
2. An explanation of what packages to install

I've never experienced the bug mentioned, and today I found out it's because I've been using the openal backend the whole time.

### PR DESCRIPTION
**Pull request by pushfoo**
Closes #1274
Closes #1900

### Changes

1. Add link targets to the platformer tutorial
2. Add new Sound page to the programming guide covering usage and compatibility
3. Link the Sound page from the top-level index
 
### How to Test

1. Clone the branch locally
2. Run `pip install -I -e .[dev]` to make sure you have the dev dependencies fully installed & upgraded
3. `./make.py html` (optional, but helps prevent an occasional rebuild loop)
4. `./make.py serve`
5. Open https://localhost:8000 in your browser
6. Verify the page is in the sidebar and Programming Guide homepage section
7. Open http://localhost:8000/programming_guide/sound.html
8. Proof-read the Sound Basics sections brutally:
    * http://localhost:8000/programming_guide/sound.html#sound-basics-loading
    * http://localhost:8000/programming_guide/sound.html#sound-basics-playing
    * http://localhost:8000/programming_guide/sound.html#sound-basics-stopping
9. Verify that [#1900](https://github.com/pythonarcade/arcade/issues/1900) is covered by http://localhost:8000/programming_guide/sound.html#backends-determine-playback-features 
10. If time / interest allow, proof read and verify other parts

### Follow-up work for future PRs

1. Move some of the preloading concepts to a new performance page / heading
2. Add an easier to digest reference counting section somewhere and link it instead of the Python contributor doc
3. Move some compatibility information to its own page / heading

### CODE REVIEW

**Comment by pushfoo** [comment]
@DavidEBrumbaugh Since you were the initial reporter for https://github.com/pyglet/pyglet/issues/952, would you be up for building the doc locally and previewing it? I've updated the test steps with instructions for doing so.

File: doc/programming_guide/sound.rst [hunk 1]
```diff
@@ -0,0 +1,724 @@
...
+
+Loading Sounds
+^^^^^^^^^^^^^^
+
+Before you can play a sound, you need to load its data into memory.
+
+Arcade provides two ways to do this. Both accept the same arguments and
+return an :py:class:`arcade.Sound` instance.
+
+The easiest way is :py:func:`arcade.load_sound`::
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst]: Could we update this (and the others) to use this method of listing code samples?

```
.. code-block:: python

    print("Hello World!")
```

File: doc/programming_guide/sound.rst [hunk 2]
```diff
@@ -0,0 +1,724 @@
...
+argument of :py:func:`arcade.play_sound`::
+
+    # Important: this *must* be a Sound instance, not a path or string!
+    self.hurt_player = arcade.play_sound(hurt_sound)
+
+Both return a :py:class:`pyglet.media.player.Player`. You should store
+it somewhere if you want to be able to stop or alter a specific playback of
+a :py:class:`~arcade.Sound`'s data.
+
+Sounds vs pyglet Players
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst]: Confusing? Maybe:
The arcade sound class and pyglet players

File: doc/programming_guide/sound.rst [hunk 3]
```diff
@@ -0,0 +1,724 @@
...
+.. _sound-basics-stopping:
+
+Stopping Sounds
+^^^^^^^^^^^^^^^
+
+
+Arcade's helper functions are the easiest way to stop playback. To use them, do
+the following:
+
+#. Choose one of the following:
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst]: Having 'following' in here twice is a bit confusing. Any way we could tighten this language?

File: doc/programming_guide/sound.rst [hunk 4]
```diff
@@ -0,0 +1,724 @@
...
+.. [#meaningbestformatheader]
+   See :ref:`sound-compat-easy` to learn more.
+
+.. [#staticsourcefoot]
+   See the :py:class:`pyglet.media.StaticSource` class used by arcade.
+
+.. _sound-loading-modes-static:
+
+Static Sounds Can Be Fastest
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst:261]: Static sounds can be faster than streaming sounds

File: doc/programming_guide/sound.rst [hunk 5]
```diff
@@ -0,0 +1,724 @@
...
+
+As long as you have enough memory, preloading entire sounds prevents
+in-game slowdowns.
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst:264]: Maybe call out the memory part a bit more. This is why we don't put an entire song in memory as static.

File: doc/programming_guide/sound.rst [hunk 6]
```diff
@@ -0,0 +1,724 @@
...
+
+This is because disk access is one of the slowest things a computer can
+do. Avoiding it during gameplay is important if your gameplay needs to
+be fast and smooth.
+
+Any of the following suggest a sound should be loaded as a static effect:
+
+* You need to start playback quickly in response to gameplay.
+* 2 or more "copies" of the sound can be playing at the same time.
```
Comment (pvcraven) [inline doc/programming_guide/sound.rst]: Spell out words less than ten.

**Review by pvcraven — CHANGES_REQUESTED** [review]

**Comment by DavidEBrumbaugh** [comment]
This thing popped up back into my inbox I didn't see it initially I apologize if you still need me to look into it please let me know

⁣Get BlueMail for Android ​

File: doc/programming_guide/sound.rst (hunk 2 above)
Comment (pushfoo) [reply doc/programming_guide/sound.rst]: I've tried to fix it for now using alternate phrasing, but this is something which might improve upstream with time.

TL;DR: There's discussion of renaming `pyglet.media.Player` to `pyglet.media.AudioPlayer` in 

> I've thought about potentially renaming the AudioPlayer classes to something more generic, so it's not confusing with the high level Player.
> -Benjamin
[*Source: Discord message*](https://discord.com/channels/438622377094414346/1162528803373977681/1164140154328465458)

> I'm going to push out a 2.10 release first, just due to the number of changes here. If all goes well, we can get this into the next one. I'll try to give it another look soon, but at a first review it's already in pretty good shape.
> -Benjamin
[*Source: Discord message*](https://discord.com/channels/438622377094414346/1162528803373977681/1165123439418408990)

This may release in 2.1.1 or 2.2 since there are a lot of changes involved, but it looks like we can expected to deleted this section eventually.

File: doc/programming_guide/sound.rst (hunk 4 above)
Comment (pushfoo) [reply doc/programming_guide/sound.rst:261]: I would like to use a longer title, but this will start getting ugly in the Furo sidebars. Were' already at the line length limit for the right sidebar:

Current:
image
Proposed:
image

Comment (pushfoo) [reply doc/programming_guide/sound.rst:261]: Marking this item as resolved since it got a thumbs up on Discord.

**Comment by pushfoo** [comment]
@DavidEBrumbaugh 

I would appreciate it if you could verify the PulseAudio crash you reported is adequately covered by the `Backends Determine Playback Features` heading.

The doc about the crash will be relevant for another few months at the minimum. Although it seems like there is a fix in the works, it sounds like it may be delayed until pyglet 2.2.

If you're travelling without access to a fast connection, this may be the easiest approach:

1. Wait for this to merge
2. Wait for the readthedocs build to complete & add the new page
3. Once the build finishes, check the `Backends Determine Playback Features` heading at https://api.arcade.academy/en/development/programming_guide/sound.html (won't exist until this is merged & built)

**Review by pvcraven — APPROVED** [review]
